### ISSUE TITLE
fetchAll behavior seems to be different in debug vs production mode

### ISSUE THREAD

**Issue by thomasmcgee** [cakephp/cakephp#7264]
Cake 3.0.11
PHP 5.6.10
The following code seems to return different results based on whether app.php debug is true or false:

        $conn = \Cake\Datasource\ConnectionManager::get('default');
        $values = $conn->execute($query)->fetchAll();
        return $values;

app.php [debug=false] returns indexed arrays:
    [0] => 318222454
    [1] => 3.0
    [2] => false

app.php [debug=true] returns associative arrays:
    [id] => 318222454
    [total_hours] => 3.0
    [billable] => false

-----------------------------------
Setting fetchAll('assoc') returns associated arrays in both debug and production mode as expected.

        $conn = \Cake\Datasource\ConnectionManager::get('default');
        $values = $conn->execute($query)->fetchAll();
        return $values;

**Comment by lorenzo** [comment]
the default of fetchAll() is PDO::FETCH_BOTH so you will have bot numeric and associative arrays as a result. How are you debugging the array when debug = false?

**Comment by thomasmcgee** [comment]
I was debugging with         \Cake\Log\Log::write(LOG_DEBUG, $values);
log output shows associative arrays in debug and indexed arrays in prod.

**Comment by markstory** [comment]
@thomasmcgee What class is returned from `execute()` with debug on/off? Does it change?

@lorenzo All of the built-in statement classes have a default value of 'num' not 'assoc'.

**Comment by thomasmcgee** [comment]
@markstory Hi, I ended up refactoring away from this, but I tried reverting code to see if I could get you an answer. On my first couple of passes last night I wasn't able to reproduce the issue. 
When I get a chance I'm going to check out my previous commit and reproduce it with the original code.

The fact that I can't reproduce it now leads me to think I might have had a cached model object that was not acting as expected. Would it be possible that the cached model object would work differently with debug on or off?
